compiler/rustc_type_ir/src/sty.rs

```diff
@@ -186,6 +186,42 @@ pub enum TyKind<I: Interner> {
     Error(I::DelaySpanBugEmitted),
 }
 
+#[allow(rustc::usage_of_ty_tykind)]
+impl<I: Interner> Clone for TyKind<I> {
```

Comment:
This is because `I: Clone` doesn't hold, isn't it?

Reply:
Yup